snowshu/core/graph.py

```diff
@@ -70,6 +69,58 @@ def catalog_difference(source_graph: Union["SnowShuGraph", networkx.Graph],
         source_graph.remove_nodes_from(nodes_to_delete)
         return source_graph
 
+    @staticmethod
+    def _build_graph_cycles(graph_cycles: list) -> Tuple[str, str]:
+        """ Builds simple cycles in a directed graph per replica config.
+
+            Args:
+                configs: :class:`Configuration <snowshu.core.configuration_parser.Configuration>` object.
+
+            Returns:
+                The :rtype: Tuple[str, str]: `message`, `filename`,
+                the `message` represents the list of the simple cycles in the graph,
+                the `filename` corresponds to the name of .png and .graphml files where cycles image is saved.
+        """
+
+        cycle_graph = networkx.DiGraph()
+        message = ""
+        nodes = []
+
+        # create a message with list of nodes for each simple cycle, add nodes and edges to the graph
+        for cycle in graph_cycles:
+            for i, node in enumerate(cycle):
+                nodes.append(node)
+                if i != len(cycle) - 1:
+                    cycle_graph.add_edge(node, cycle[i + 1])
+                    message = message + '\033[1;34m' + str(node)[17:-1] + '\t\033[1;32m----\t'
+                else:
+                    cycle_graph.add_edge(node, cycle[0])
+                    message = message + '\033[1;34m' + str(node)[17:-1]
+            message = message + '\n\t'
+
+        # label each node as DataBase.Schema.Table
+        nodes = set(nodes)
+        label_dict = {}
+        for node in cycle_graph.nodes():
+            label_dict[node] = str(node)[17:-1].replace('.', '\n')
+
+        # make an image for the existing simple cycles
+        created_at = datetime.now()
+        plt.figure(figsize=(8, 8))
+        plt.margins(0.1)
+        plt.title(f'\nGraph of the simple cycles, created at: {created_at.strftime("%d/%m/%Y %H:%M:%S")}')
+
+        networkx.draw(
+            cycle_graph, pos=networkx.planar_layout(cycle_graph), labels=label_dict, with_labels=True,
+            node_size=1000, node_color='skyblue', font_size=6, font_color='green', width=2,
+            horizontalalignment='center', verticalalignment='center', connectionstyle='arc3, rad=0.05')
+
+        filename = f'{GraphSetRunner.barf_output}/{created_at.strftime("%d_%m_%Y_%H_%M_%S")}'
```

Comment:
We should probably name the files appropriately instead of just using a datetime.

Nitpick: The timestamp should also be reordered year-month-day instead of day-month-year. Same for the graph title. It should hopefully avoid any confusion on timestamp format.